libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -996,3 +965,61 @@ export interface SimulatorServerResponse {
   /** The error that occurred during the method call. */
   readonly error?: any;
 }
+
+function planUpdate(current: BaseResourceSchema[], next: BaseResourceSchema[]) {
+  const currentByPath = resourceByPath(current);
+  const nextByPath = resourceByPath(next);
+
+  const added: string[] = [];
+  const updated: string[] = [];
+  const deleted: string[] = [];
+  const retain: string[] = [];
+
+  for (const [path, nextConfig] of Object.entries(nextByPath)) {
+    const currConfig = currentByPath[path];
+
+    // if the resource is not in "current", it means it was added
+    if (!currConfig) {
+      added.push(nextConfig.path);
+      continue;
+    }
+
+    // the resource is already in "current", if it's different from "next", it means it was updated
+    const state = (r: BaseResourceSchema) =>
+      JSON.stringify({
+        props: r.props,
+        type: r.type,
+      });
```

Comment:
The dependent *are* taken into account but not here. If a resource is stopped (either because of an update or a delete), all of it's dependents are also stopped. Makes sense?

Reply:
Added a comment.

Reply:
yes, that makes sense.

I only want to mention that `cloud.Deploy` has a semantics of "run this code every time my app is updated / I do a new deployment" and I'm not sure if `simulator.update()` is respecting this

Reply:
Ok, node deps are now taken into account (added a test).